bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/WebClientFactoryImpl.java

```diff
@@ -217,7 +241,44 @@ public HttpClient run() {
                 throw (RuntimeException) cause;
             } else {
                 throw new HttpClientInitializationException(
-                        "unexpected checked exception during initialization of the jetty client", cause);
+                        "unexpected checked exception during initialization of the Jetty http client", cause);
+            }
+        }
+    }
+
+    private WebSocketClient createWebSocketClientInternal(String consumerName, String endpoint, boolean startClient) {
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<WebSocketClient>() {
+                @Override
+                public WebSocketClient run() {
+                    logger.debug("creating web socket client for endpoint {}", endpoint);
+                    SslContextFactory sslContextFactory = createSslContextFactory(endpoint);
+
+                    WebSocketClient webSocketClient = new WebSocketClient(sslContextFactory);
+                    final QueuedThreadPool queuedThreadPool = createThreadPool(consumerName, minThreadsCustom,
+                            maxThreadsCustom, keepAliveTimeoutCustom);
+
+                    webSocketClient.setExecutor(queuedThreadPool);
+
+                    if (startClient) {
+                        try {
+                            webSocketClient.start();
```

Comment:
in `deactivate` the commonHttpClient is stopped, so should the new webSocketClient.